### PR TITLE
add authentication with JWT

### PR DESCRIPTION
**Pull request by bravint**
run npm ci after merge

### CODE REVIEW

File: src/server/utils/joi.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        .required(),
+    password: joi.string()
+        .min(6)
+        .required(),
+    //repeat_password: joi.ref('password'),
+    email: joi.string().
+        email({ 
+            minDomainSegments: 2,
+            tlds: { allow: true } 
+        }),
```
Comment (nathanSamosa) [inline src/server/utils/joi.js:16]: RIP nonsense emails

**Review by nathanSamosa — APPROVED** [review]

**Review by molloyconor511 — APPROVED** [review]

**Review by polarbear23 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -11,7 +11,11 @@
     "cors": "^2.8.5",
     "dotenv": "^16.0.0",
     "express": "^4.17.3",
+    "joi": "^17.6.0",
```
Comment (glowkeeper) [inline package.json:14]: Ooo - I'm a massive Joi fan :) Schema validation - nice would like to see _exactly_ what you're using that for...

File: src/server/controllers/admin.js [hunk 3]
```diff
@@ -1,52 +1,75 @@
...
-  }
+const { checkPassword, hashPassword } = require('../utils/bcrypt');
+const { registerSchema, loginSchema } = require('../utils/joi');
+const { createToken } = require('../utils/jwt');
+const { prisma } = require('../utils/prisma');
+
+const { HTTP_RESPONSE } = require('../config');
+
+const createAdmin = async (req, res) => {
+    const { error } = registerSchema.validate(req.body);
```
Comment (glowkeeper) [inline src/server/controllers/admin.js:9]: Aha! Joi :) Nice.

File: src/server/utils/joi.js (hunk 1 above)
Comment (glowkeeper) [reply src/server/utils/joi.js:16]: And Joi again - awesome

**Review by glowkeeper — APPROVED** [review]
Joi! Approved :)

File: package.json (hunk 2 above)
Comment (bravint) [reply package.json:14]: Hopefully to reject badly formed requests before they get to Prisma and bring the server down

**Pull request merged**